pandas/core/window/rolling.py

```diff
@@ -656,9 +656,28 @@ def _numba_apply(
             return self._resolve_output(out, obj)
 
     def aggregate(self, func, *args, **kwargs):
-        result = ResamplerWindowApply(self, func, args=args, kwargs=kwargs).agg()
+        # GH46132
+        # modifying axis and transposing dataframe should not be needed
+        # once ReamplerWindow supports axis = 1
+
+        obj = self.obj
+        axis = self.axis
+
+        self.obj = self.obj.T if self.axis == 1 else self.obj
+        self.axis = 0
+
+        try:
+            result = ResamplerWindowApply(self, func, args=args, kwargs=kwargs).agg()
```

Comment:
can we just add an axis argument here and thread it thru. this state change is not at all obvious.

Reply:
I just would like to make sure if you want me to make this comment more clear? And I notice that in PR #47078, it says rolling window doesn't support axis = 1 anymore and that caused this change didn't pass the test.

Reply:
i am not sure what you mean. I don't really like this approach because its hiding the axis state. What I am suggesting is we actually fix this by passing thru the axis argument to do this properly.

Reply:
Ok, the `agg` method of  `ResamplerWindowApply` hasn't been implemented, yet, and it will call the `agg` method of `Apply` class which can't recognize that we want to agg on axis=1 instead of axis=0. Even though I am not entirely familiar with pandas, I think passing the axis argument could not help.